flocker/node/agents/ebs.py

```diff
@@ -371,16 +370,22 @@ def _get_ebs_volume(self, blockdevice_id):
 
         :param unicode blockdevice_id: ID of a blockdevice that needs lookup.
 
-        :returns boto.ec2.volume.Volume for the input id. ``None`` if
-            no boto.ec2.volume.Volume was found for the given id.
+        :returns boto.ec2.volume.Volume for the input id.
+
+        :raise UnknownVolume: If no volume with a matching identifier can be
+             found.
         """
-        for volume in self.connection.get_all_volumes(
-                volume_ids=[blockdevice_id]):
-            if volume.id == blockdevice_id:
-                # Sync volume for uptodate metadata
-                volume.update()
-                return volume
-        return None
+        try:
+            for volume in self.connection.get_all_volumes(
+                    volume_ids=[blockdevice_id]):
+                if volume.id == blockdevice_id:
+                    return volume
+        except EC2ResponseError as e:
+            if e.error_code == "InvalidName.NotFound":
```

Comment:
Ugh, classy.

Maybe worth a link to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#CommonErrors since it took me a few minutes to find.